### ISSUE TITLE
Suggestion: Auto-select my kingdom if there is only one left

### ISSUE THREAD

**Issue by d-albrecht** [Sesu8642/FeudalTactics#33]
No matter how you play the game, it is highly likely that at some point you will have exactly one kingdom. Whether this is by combining all kingdoms or getting some destroyed by others. Would it be possible to automatically select this single remaining kingdom as soon as it is my turn, such that I can immediately click on "buy unit" or "buy castle"?

It is possible that you "get back" to having several kingdoms afterwards, and hence, the automation has to check each round/turn if it should trigger. But I see no obvious reason to have the player do an additional yet unnecessary mouse click / display touch. And if you want to select the kingdom manually, nothing will stop you from doing so, it will just have no real effect to select the only remaining and already selected kingdom.

**Comment by Sesu8642** [comment]
Yes, that makes sense. I previously thought of remembering and re-selecting the kingdom that was active in the previous turn but that turned out to be rather complicated. In the case where there is only one left, it should be pretty straight forward.

**Comment by d-albrecht** [comment]
I just realized, if you have only one kingdom left and the kingdom gets auto-selected, this might mess with the warning dialog about "forgotten kingdoms" in the way that you could hit the "end turn" button without doing anything and the game might count that as a turn (which you might not have intended) or warn you (when you knowingly didn't want to change anything). So, you might have to come up with what suits you more. Both solutions have their drawbacks. But either way I would prefer to not have to select the only remaining kingdom. Personally, I would probably prefer that you get a warning if you didn't do anything this turn, but as far as I have already looked into the code, this might be the more difficult approach code-wise.

**Comment by Sesu8642** [comment]
I also prefer showing the warning in that situation. The Kingdom#setWasActiveInCurrentTurn would need to be called every time the player does any action in the kingdom and just like now when it is activated. In that one case where it is auto-activated, that would need to be skipped or set back to false again.
